engines/order_management/app/services/order_management/order/updater.rb

```diff
@@ -162,6 +163,12 @@ def after_payment_update(payment)
 
       private
 
+      def cancel_payments_requiring_auth
+        return unless order.payment_state == "paid"
```

Comment:
This is checked above already.

Reply:
They're actually two different conditions. I could move them both down into this method, but I'd have to pass in the `last_payment_state` here, like:
```ruby
  def cancel_payments_requiring_auth(last_payment_state)
    return unless order.payment_state == "paid" && last_payment_state != "paid"
    ...
```
Would that be preferable?

Reply:
Ah, now I see. I overlooked `order.payment_state = infer_payment_state` in between. :see_no_evil: 

Good to go.